ultradns-rest/src/main/java/denominator/ultradns/UltraDNSRestProvider.java

```diff
@@ -0,0 +1,187 @@
+package denominator.ultradns;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+
+import javax.inject.Singleton;
+
+import dagger.Provides;
+import denominator.BasicProvider;
+import denominator.CheckConnection;
+import denominator.DNSApiManager;
+import denominator.QualifiedResourceRecordSetApi;
+import denominator.ResourceRecordSetApi;
+import denominator.ZoneApi;
+import denominator.config.ConcatBasicAndQualifiedResourceRecordSets;
+import denominator.config.NothingToClose;
+import denominator.config.WeightedUnsupported;
+import denominator.profile.GeoResourceRecordSetApi;
+//import UltraDNSRestErrorDecoder.UltraDNSError;
+import feign.Feign;
+import feign.Logger;
+import feign.Request.Options;
+import feign.form.FormEncoder;
+import feign.gson.GsonDecoder;
+import feign.gson.GsonEncoder;
+
+import static dagger.Provides.Type.SET;
+
+public class UltraDNSRestProvider extends BasicProvider {
+
+  private final String url;
+
+  public UltraDNSRestProvider() {
+    this(null);
+  }
+
+  /**
+   * @param url if empty or null use default
+   */
+  public UltraDNSRestProvider(String url) {
+    this.url =
+            url == null || url.isEmpty() ? "https://test-restapi.ultradns.com/v2" : url;
```

Comment:
To make the code cleaner, I think its better to use the utility method from guava or apache for the null/empty checks.